lib/sparql_to_sw_solr/language_field.rb

```diff
@@ -0,0 +1,42 @@
+require_relative 'searchworks_languages'
+
+module SparqlToSwSolr
+  class InstanceSolrDoc
+    module LanguageField
+
+      private
+
+      def language_values
+        result = []
+        solution_values(language_solutions).each do |soln_val|
+          # TODO: ultimately, should lookup lang uri's RDF and retrieve it (and cache it) and then
+          # get value from "<skos:prefLabel xml:lang="en">" property (or <madsrdf:authoritativeLabel xml:lang="en">)
+          code = soln_val[%r{^http://id.loc.gov/vocabulary/languages/(.*)}, 1]
+          next unless code && !code.empty?
+          value = SEARCHWORKS_LANGUAGES[code]
+          result << (value ? value : code)
+        end
+        result
+      end
```

Comment:
Extract the "language lookup" into another method to be called in this iteration.  Then that method can change later without changing this iteration method and the iteration can use a compact form, like `[].map`